MonMaraicher/Features/Map/MapViewModel.swift

```diff
@@ -28,12 +28,11 @@ final class MapViewModel: ObservableObject {
     }
 
     private func loadFarmers() {
-        farmerService.loadFarmers { result in
-            switch result {
-            case .success(let farmers):
-                self.allFarmers = farmers
-            case .failure:
-                print(LoadFarmerError.errorDecoding)
+        DispatchQueue.main.async {
+            do {
+                self.allFarmers = try self.farmerService.loadFarmers()
+            } catch {
+                print("Error decoding: \(FarmerService.Error.decodingError)")
```

Comment:
You are forcing to print the decoding error here... If the error was `invalidJsonFile` how to know it by doing that? Why not printing the error you get with the catch? :) 
The following code should work ;)

```
print("Error when loading farmers: \(error)"
```

Reply:
I asked myself the same question when I wrote this code. The JSON file error was not present and it was a mistake IMO
Your suggestion is better because it returns the right error message depending on the case